runtime/src/bank.rs

```diff
@@ -1647,10 +1659,23 @@ impl Bank {
         let accounts_delta_hash = self.rc.accounts.bank_hash_info_at(self.slot());
         let mut signature_count_buf = [0u8; 8];
         LittleEndian::write_u64(&mut signature_count_buf[..], self.signature_count() as u64);
+
+        let mut hard_fork_buf = [0u8; 8];
+        LittleEndian::write_u64(
+            &mut hard_fork_buf[..],
+            *self
+                .hard_forks
+                .read()
+                .unwrap()
+                .get(&self.slot())
+                .unwrap_or(&0),
+        );
+
         let hash = hashv(&[
             self.parent_hash.as_ref(),
             accounts_delta_hash.hash.as_ref(),
             &signature_count_buf,
+            &hard_fork_buf,
```

Comment:
This causes all ledgers created by the 0.22 version line earlier this PR to be incompatible with the ones created by this PR. Is it intentional? Or, only hash hard_fork_buf if it's not 0? That would preserve the compatibility on the other hand.

Reply:
good point, done